charts/mattermost-rtcd/templates/daemonset.yaml

```diff
@@ -55,10 +55,14 @@ spec:
               containerPort: {{ .Values.service.APIport }}
               hostPort: {{ .Values.service.APIport }}
               protocol: TCP              
-            - name: rtc
+            - name: rtc-udp
               containerPort: {{ .Values.service.RTCport }}
```

Comment:
this won't work as we use the same port and we have the `hostPort`. The ports will overlap each other.

Reply:
I tested a bit locally and could get it to expose it this way. The only caveat was that minikube doesn't support mixed protocols on anything but port `443` so I used that one (https://minikube.sigs.k8s.io/docs/tutorials/nginx_tcp_udp_ingress/#caveats).

```
Name:         mattermost-rtcd-n8gx5
Namespace:    default
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Wed, 21 Jun 2023 10:41:53 -0600
Labels:       app.kubernetes.io/instance=mattermost-rtcd
              app.kubernetes.io/name=mattermost-rtcd
              controller-revision-hash=5bc89cc7f5
              pod-template-generation=1
Annotations:  <none>
Status:       Running
IP:           192.168.49.2
IPs:
  IP:           192.168.49.2
Controlled By:  DaemonSet/mattermost-rtcd
Containers:
  mattermost-rtcd:
    Container ID:   docker://74cfd07566702e8693c120a8ca9b9bba74700235698ad7abdac1942ebac9a245
    Image:          mattermost/rtcd-daily:dev-abfdcea
    Image ID:       docker-pullable://mattermost/rtcd-daily@sha256:84dc49cfe6fc5e578b37ca8b7338456ec59fe6fb77a297cc42c1f0b531da5480
    Ports:          8045/TCP, 443/UDP, 443/TCP
    Host Ports:     8045/TCP, 443/UDP, 443/TCP
    State:          Running
      Started:      Wed, 21 Jun 2023 10:41:53 -0600
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     1
      memory:  2Gi
    Requests:
      cpu:      100m
      memory:   32Mi
    Liveness:   http-get http://:api/version delay=0s timeout=1s period=10s #success=1 #failure=3
    Readiness:  http-get http://:api/version delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      RTCD_API_SECURITY_ALLOWSELFREGISTRATION:  true
      RTCD_LOGGER_CONSOLELEVEL:                 DEBUG
      RTCD_LOGGER_ENABLEFILE:                   false
      RTCD_RTC_ICEPORTTCP:                      443
      RTCD_RTC_ICEPORTUDP:                      443
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-d69tz (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  kube-api-access-d69tz:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/network-unavailable:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists
                             node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                             node.kubernetes.io/unreachable:NoExecute op=Exists
                             node.kubernetes.io/unschedulable:NoSchedule op=Exists
                             rtcd=true:NoSchedule
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  8s    default-scheduler  Successfully assigned default/mattermost-rtcd-n8gx5 to minikube
  Normal  Pulled     9s    kubelet            Container image "mattermost/rtcd-daily:dev-abfdcea" already present on machine
  Normal  Created    9s    kubelet            Created container mattermost-rtcd
  Normal  Started    9s    kubelet            Started container mattermost-rtcd
```

But more importantly, do we actually need to expose the ports there when `hostNetwork` is `true`? My very basic understanding is that we don't but would like to hear from you guys.

Reply:
@spirosoik @stylianosrigas Not urgent but would be nice to sort this out whenever you get a chance so that eventually we could get it working on Community as we had a few people unable to join in the past due to client-side UDP restrictions.

Also happy to jump on a call if necessary, just let me know :)